### ISSUE TITLE
It should be possible to provide a file to LFS_JWT_SECRET

### ISSUE THREAD

**Issue by varanauskas** [go-gitea/gitea#25034]
### Feature Description

**TLDR:** `LFS_JWT_SECRT` should be able to be stored in a separate file outlined by outlined by `LFS_JWT_SECRET_URI` 

Secrets in the `[security]` section of config allow providing both an inline secret as well as a file where the secret is stored with the `_URI` suffix.

For example
```
# /etc/gitea/app.ini
[security]
INTERNAL_TOKEN=#SOME_SECRET_VALUE

# Can be replaced by:
# /etc/bites/app.ini
[security]
INTERNAL_TOKEN_URL=/etc/gitea/secrets/internal_token

# /etc/gitea/secrets/internal_token then contains the #SOME_SECRET_VALUE
```

This allows for better installation/maintenance of Gitea, as /etc/gitea/app.ini could be stored in version control, as all secrets would be outlined separately.

## Proposal:

I propose adding a new config key `LFW_JWT_SECRET_URI` that would work the same as `INTERNAL_TOKEN_URI` by allowing users to specify a separate file where the `LFS_JWT_SECRET` is stored

**Comment by varanauskas** [comment]
I do not have Go experience, but could try implementing this as it seems simple enough, as long as someone can review my changes

**Comment by techknowlogick** [comment]
@lonix1 two have the URI as an option.

As well, if you are using the docker image, then all of them are, in theory, available as reading from a file if using env vars.

**Comment by wxiaoguang** [comment]
Make environment-to-ini support loading key value from file #24832

**Comment by wxiaoguang** [comment]
It's for docker users. See the documents.

```
	Environment variables of the form "GITEA__SECTION_NAME__KEY_NAME__FILE"
	will be mapped to the ini section "[section_name]" and the key
	"KEY_NAME" with the value loaded from the specified file.
```

And https://docs.gitea.com/1.20/installation/install-with-docker#managing-deployments-with-environment-variables

**Comment by wxiaoguang** [comment]
> keeping secrets in environment variables is a bad idea. Sometimes it's the only option though.

Sorry I didn't get your point. See:

> Environment variables of the form "GITEA__SECTION_NAME__KEY_NAME__FILE"
> with the value loaded from the specified file.

They are indeed able to be stored in your file.

**Comment by wxiaoguang** [comment]
Yes, it should work this way.

If it doesn't work, feel free to open an issue for it (with reproducible steps)

**Comment by wxiaoguang** [comment]
> The environment-to-ini option is not suitable. It is not meant for secrets - it will hardcode the secrets into `app.ini` which will mean they are readable by anyone on the host.

I can understand your meaning but I can't understand why it is a problem.

The `app.ini` is 0600 by default:

```
$ ls -l custom/conf/app.ini
-rw------- 1 git git 2839 Jun 16 20:35 custom/conf/app.ini
```

I do not understand why `they are readable by anyone on the host`.

If there is anyone who can read `app.ini`, they can also read your "secret files" at the same time.

**Comment by varanauskas** [comment]
Thank you, environment-to-ini seems to fit my use case and `_URI` endpoints seem to not be needed anymore. However, these two approaches might seem confusing to new users, perhaps a best practice demonstration in docs on how to decide which one to choose would be beneficial. That’s probably something for gitea developers to decide

**Comment by varanauskas** [comment]
> (I have a compromise solution which keeps secrets out of version control, but I'm using automation (ansible). I'm unsure of your approach to deployment.)

Exactly I also use Ansible

**Comment by wxiaoguang** [comment]
Actually, unlike other secrets, this "LFS_JWT_SECRET" do not need to be pre-generated.

You can keep it empty, then when Gitea starts, the secret will be generated automatically.

**Comment by varanauskas** [comment]
Is it okay if LFS_JWT_SECRET is different on each boot with Same db and sessions?

**Comment by techknowlogick** [comment]
I've implemented the other secrets to be able to be loaded via URI in the linked PR. If anyone want to test it that would be appreciated :)

**Comment by techknowlogick** [comment]
@lonix1 the nightly version would have it. either the binary https://dl.gitea.com/gitea/main/ or the `:nightly` docker image would be where you can run it from.
